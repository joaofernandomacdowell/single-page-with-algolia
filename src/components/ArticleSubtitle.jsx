import React from 'react';

const ArticleSubtitle = ({ subtitle }) =>
  <p className="zoom-article-subtitle">{subtitle}</p>

export default ArticleSubtitle;