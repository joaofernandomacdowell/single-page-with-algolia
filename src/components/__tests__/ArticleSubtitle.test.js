import React from 'react'
import { shallow } from 'enzyme';

import mockApiResponse from '../../helper';
import ArticleSubtitle from '../ArticleSubtitle';


describe('ArticleSubtitle component', () => {
  const props = {
    subtitle: mockApiResponse.hits[0].subtitle
  };
  const wrapper = shallow(<ArticleSubtitle {...props} />);

  test('match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('have subtitle label with correct text', () => {
    const label = wrapper.find('.zoom-article-subtitle');

    expect(label).toHaveLength(1);
    expect(label.text()).toBe(mockApiResponse.hits[0].subtitle);
  });
});