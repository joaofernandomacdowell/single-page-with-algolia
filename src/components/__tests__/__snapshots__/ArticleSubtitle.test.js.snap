// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleSubtitle component match the snapshot 1`] = `ShallowWrapper {}`;
